### PR TITLE
add program check

### PR DESCRIPTION
**Pull request by panyx0718**
test=develop

### CODE REVIEW

File: paddle/fluid/framework/ir/graph.cc [hunk 1]
```diff
@@ -23,8 +23,62 @@ limitations under the License. */
...
 namespace ir {
+namespace {
+
+void CheckProgram(const ProgramDesc &program) {
+  std::map<int, bool> visit;
+#define _INT(role) static_cast<int>(role)
+
+  for (size_t i = 0; i < program.Size(); ++i) {
+    for (OpDesc *op : program.Block(i).AllOps()) {
+      // For backward compatibility, some program doesn't have role added.
```
Comment (chengduoZH) [inline paddle/fluid/framework/ir/graph.cc:34]: Which program doesn't have role?

**Review by chengduoZH — COMMENTED** [review]
Do you run this PR in CE models?

File: paddle/fluid/framework/ir/graph.cc (hunk 1 above)
Comment (panyx0718) [reply paddle/fluid/framework/ir/graph.cc:34]: program generated from old paddle version

**Review by chengduoZH — APPROVED** [review]
LGTM

**Pull request merged**